synapse/handlers/set_password.py

```diff
@@ -48,9 +55,12 @@ async def set_password(
         logout_devices: bool,
         requester: Optional[Requester] = None,
     ) -> None:
-        if not self._auth_handler.can_change_password():
+        if self._auth_handler.can_change_password():
```

Comment:
We should keep the `not` here :)

```suggestion
        if not self._auth_handler.can_change_password():
```

Reply:
Woops, you're right!